backend/model/model.go

```diff
@@ -1590,17 +1590,38 @@ func MigrateDB(ctx context.Context, DB *gorm.DB) (bool, error) {
 	// ignore errors - an error means that there are no partitions, so we can safely use the zero-value.
 	DB.Raw("select split_part(relname, '_', 5) from pg_stat_all_tables where relname like 'error_object_embeddings_partitioned%' order by relid desc limit 1").Scan(&lastCreatedPart)
 
-	// Make sure partitions are created for the next 1k projects
-	for i := lastCreatedPart; i < lastVal+1000; i++ {
-		sql := fmt.Sprintf(`
+	// Make sure partitions are created for the next 1k projects, starting with the next partition needed
+	for i := lastCreatedPart + 1; i < lastVal+1000; i++ {
+		if err := DB.Exec(fmt.Sprintf(`
 			CREATE TABLE IF NOT EXISTS error_object_embeddings_partitioned_%d
 			PARTITION OF error_object_embeddings_partitioned
 			FOR VALUES IN ('%d');
-		`, i, i)
-
-		if err := DB.Exec(sql).Error; err != nil {
+		`, i, i)).Error; err != nil {
 			return false, e.Wrapf(err, "Error creating partitioned error_object_embeddings for index %d", i)
 		}
+
+		if err := DB.Exec(fmt.Sprintf(`
+			CREATE INDEX ON error_object_embeddings_partitioned_%d
+			USING ivfflat (gte_large_embedding vector_l2_ops) WITH (lists = 1000);
+		`, i)).Error; err != nil {
+			return false, e.Wrapf(err, "Error creating index error_object_embeddings for index %d", i)
+		}
+
+		// in case this partition was already attached by a previous failed migration, try detach first
```

Comment:
to minimize locking on the table, would probably be better to only attach partition and ignore any "partition already attached" error